authui/src/countdown.ts

```diff
@@ -0,0 +1,87 @@
+import { Controller } from "@hotwired/stimulus";
+import { Duration } from "luxon";
+
+export class CountdownController extends Controller {
+  static targets = ["button"];
+  static values = {
+    cooldown: Number,
+    cooldownUntil: String,
+    label: String,
+    labelUnit: String,
+    format: String,
+  };
+
+  declare readonly buttonTarget: HTMLButtonElement;
+
+  declare cooldownValue?: number;
+  declare cooldownUntilValue?: string;
+  declare labelValue: string;
+  declare labelUnitValue: string;
+  declare formatValue?: string;
+
+  animHandle: number | null = null;
+
+  connect() {
+    this.validateValues();
+
+    const button = this.buttonTarget;
+
+    const scheduledAt = new Date();
+    const cooldown = this.cooldownValue
+      ? this.cooldownValue * 1000
+      : new Date(this.cooldownUntilValue!).getTime() - scheduledAt.getTime();
+    const label = this.labelValue;
+    const labelUnit = this.labelUnitValue;
+    const format = this.formatValue || "s";
+
+    const tick = () => {
+      const now = new Date();
+      const timeElapsed = now.getTime() - scheduledAt.getTime();
+
+      let seconds = 0;
+      if (timeElapsed <= cooldown) {
+        seconds = Math.round((cooldown - timeElapsed) / 1000);
+      }
+      const duration = Duration.fromObject({ seconds });
+      const formatted = duration.toFormat(format);
+
+      if (seconds === 0) {
+        button.disabled = false;
+        button.textContent = label;
+        this.animHandle = null;
+      } else {
+        button.disabled = true;
+        button.textContent = labelUnit
+          .replace("%d", String(seconds))
```

Comment:
I think we don't need to display the raw seconds. Always display a formatted duration, and give a default format to the format value which only displays "mm:ss"